jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcCrudRepository.java

```diff
@@ -0,0 +1,436 @@
+package io.kestra.jdbc.repository;
+
+import io.kestra.core.queues.QueueService;
+import io.kestra.core.repositories.ArrayListTotal;
+import io.kestra.core.utils.ListUtils;
+import io.micronaut.data.model.Pageable;
+import org.jooq.*;
+import org.jooq.Record;
+import org.jooq.impl.DSL;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Base JDBC repository for CRUD operations.
+ * <p>
+ * NOTE: it uses the <code>defaultFilter(tenantId)</code> for querying.
+ * If the child repository uses a default filter, it should override it.
+ * <p>
+ * For example, to avoid supporting allowDeleted:
+ * <pre>{@code
+ * @Override
+ * protected Condition defaultFilter(String tenantId) {
+ *     return buildTenantCondition(tenantId);
+ * }
+ *
+ * @Override
+ * protected Condition defaultFilter() {
+ *     return DSL.trueCondition();
+ * }
+ * }</pre>
+ *
+ * @param <T> the type of the persisted entity.
+ */
+public abstract class AbstractJdbcCrudRepository<T> extends AbstractJdbcRepository {
```

Comment:
`<T>` could not be  `<T extend HasUUID>` ? This will allow us to have the `exists/delete` base methods

Reply:
I'll open an issue for that as I think not all our entities extend HasUID but I agree they should!
If they all extend it, we would be able to get rid of the QueueService which would also be a win